packages/medusa/src/api-v2/admin/promotions/validators.ts

```diff
@@ -11,3 +29,121 @@ export class AdminGetPromotionsParams extends extendedFindParamsMixin({
   @IsOptional()
   code?: string
 }
+
+export class AdminPostPromotionsReq {
+  @IsNotEmpty()
+  @IsString()
+  code: string
+
+  @IsBoolean()
+  @IsOptional()
+  is_automatic: boolean
+
+  @IsOptional()
+  @IsEnum(PromotionType)
+  type: PromotionType
+
+  @IsOptional()
+  @IsString()
+  campaign_id: string
+
+  @IsOptional()
+  @ValidateNested()
+  @Type(() => AdminPostCampaignsReq)
+  campaign: AdminPostCampaignsReq
```

Comment:
question: can we do XOR on these

Reply:
Didn't find any native implementation from class-validator. Although we could custom validator or a conditional validation on this. The module validations will catch this though and throw a relevant error. We could get away with a simpler API validation.

Reply:
nice, didn't see we had this already! 🔥 Thanks